### PR TITLE
Typescript

### PR DESCRIPTION
**Pull request by dbslone**

### CODE REVIEW

File: frontend/packages/client/src/components/HomeHeader.tsx [hunk 1]
```diff
@@ -10,7 +10,13 @@ const stars = [
...
 ];
 
-export default function HomeHeader({ isVisible = true } = {}) {
+interface HomeHeaderProps {
+  isVisible: boolean;
+}
+
+export default function HomeHeader({
+  isVisible = true,
+}: HomeHeaderProps): JSX.Element | null {
```
Comment (germanurrus) [inline frontend/packages/client/src/components/HomeHeader.tsx:19]: Could we start declaring things like this?: 
```
export const HomeHeader: React.FC<HomeHeaderProps> = ({ isVisible = true }) => {
  return ....;
};
```

FC has null as a return type as well and makes is a bit more readable (IMO), although using JSX.Element we need to be more explicit in the types declaration (stronger types) and that's good too.

**Review by germanurrus — APPROVED** [review]
LGTM! 🚀 (left some comments)

File: frontend/packages/client/src/components/HomeHeader.tsx (hunk 1 above)
Comment (dbslone) [reply frontend/packages/client/src/components/HomeHeader.tsx:19]: Yea I'll switch to use React.FC

File: frontend/packages/client/src/react-app-env.d.ts [hunk 2]
```diff
@@ -0,0 +1 @@
+/// <reference types="react-scripts" />
```
[the remark this answers is not in the document]
Comment (dbslone) [reply frontend/packages/client/src/react-app-env.d.ts:1]: hmm thats odd the .d.ts file is in the shared-components lib. Sometimes the typescript compiler in vs code does wonky things. If you hit cmd+shift+p then enter typescript you should see a restart option

**Review by linxiao-li — COMMENTED** [review]
I think it might be helpful to compile a react typescript style guide which include examples of common patterns:
- Functional Components
- Hooks
- Context
- Event Handler

**Review by jbluks — APPROVED** [review]

**Review by germanurrus — APPROVED** [review]

**Comment by germanurrus** [comment]
@linxiao-li @mannynotfound do you have any comments about getting this PR merged?

**Review by linxiao-li — APPROVED** [review]
LGTM

**Pull request merged**